util/imagetools/create.go

```diff
@@ -266,3 +291,32 @@ func detectMediaType(dt []byte) (string, error) {
 
 	return images.MediaTypeDockerSchema2ManifestList, nil
 }
+
+func parseAnnotations(ann map[string]string) (map[string]map[string]string, error) {
+	annotationRegexp := regexp.MustCompile(`^([a-z-]+):(\S+)$`)
+	indexAnnotations := make(map[string]string)
+	manifestDescriptorAnnotations := make(map[string]string)
+	for k, v := range ann {
+		groups := annotationRegexp.FindStringSubmatch(k)
```

Comment:
Instead of using a regexp here, wdyt about using https://github.com/moby/buildkit/blob/dd0053cdce470b1355fdb0bd5a8f2b0fc506d842/exporter/containerimage/exptypes/annotations.go#L85 instead?

Obviously, it's not perfect, since we'd need to add the `annotation-` prefix here, which might make the error message a bit odd, but we could always upstream a buildkit fix later that would rework the function to have the caller remove the prefix there (so we wouldn't have to do that).

Reply:
> Instead of using a regexp here, wdyt about using https://github.com/moby/buildkit/blob/dd0053cdce470b1355fdb0bd5a8f2b0fc506d842/exporter/containerimage/exptypes/annotations.go#L85 instead?

Initially I had the same idea but we are using `:` as a separator for `type` and annotation key compared to `.` expected [here](https://github.com/moby/buildkit/blob/dd0053cdce470b1355fdb0bd5a8f2b0fc506d842/exporter/containerimage/exptypes/annotations.go#L20). So it needs more work than just appending `annotation-` prefix.

> but we could always upstream a buildkit fix later that would rework the function

Does it make sense to make buildkit parser to also handle `:` separator. It sounded specific to buildx so I feel we should keep it here. wdyt?

Reply:
Personally, I'd prefer the conversion to buildkit's form for now, so that we only have one source of truth for parsing these keys - I'm happy to follow up in buildkit to rework the logic to be a bit more reusable for this case.

I also just noticed (sorry), that we aren't handling platforms? `manifest-descriptor` supports a platform, and should only be attached to the descriptor for those platforms.

Reply:
> I'm happy to follow up in buildkit to rework the logic to be a bit more reusable for this case.

Agreed. That should be the way moving forward. I added a comment regrading using buildkit once it supports our use-case here.

> I also just noticed (sorry), that we aren't handling platforms?

Great Catch. Done.